File: src/status.ts
import type { WorktreeView } from "./types";

export function formatStatus(worktree: WorktreeView): string {
  if (worktree.availability !== "available") return worktree.availability;
  const counts = worktree.status;
  if (!counts) return "unavailable";
  const parts: string[] = [];
  if (counts.conflicted) parts.push(`${counts.conflicted} conflicted`);
  if (counts.staged) parts.push(`${counts.staged} staged`);
  if (counts.modified) parts.push(`${counts.modified} modified`);
  if (counts.untracked) parts.push(`${counts.untracked} untracked`);
  if (worktree.locked !== undefined) parts.push(worktree.locked ? `locked: ${worktree.locked}` : "locked");
  return parts.length ? parts.join(", ") : "clean";
}

File: src/status.rs
use crate::types::{Availability, Worktree};
pub fn format(worktree: &Worktree) -> String {
    match worktree.availability {
        Availability::Prunable => return "prunable".into(),
        Availability::Bare => return "bare".into(),
        Availability::Unavailable => return "unavailable".into(),
        Availability::Available => {}
    }
    let Some(c) = &worktree.status else {
        return "unavailable".into();
    };
    let mut p = Vec::new();
    if c.conflicted > 0 {
        p.push(format!("{} conflicted", c.conflicted));
    }
    if c.staged > 0 {
        p.push(format!("{} staged", c.staged));
    }
    if c.modified > 0 {
        p.push(format!("{} modified", c.modified));
    }
    if c.untracked > 0 {
        p.push(format!("{} untracked", c.untracked));
    }
    if let Some(reason) = &worktree.locked {
        p.push(if reason.is_empty() {
            "locked".into()
        } else {
            format!("locked: {reason}")
        });
    }
    if p.is_empty() {
        "clean".into()
    } else {
        p.join(", ")
    }
}
